File: JavaScript_Basic_04_2023_SolutionsOfProblems/02.MoreEx_JS_Conditional_Stitiments-More Exercises/P03_Harvest.js
function harvest(input) {

    let areaSqrM = Number(input[0]);
    let grapePerSqrMKg = Number(input[1]);
    let neededWineLtr = Number(input[2]);
    let numberWorkers = Number(input[3]);

    let producedWineLtr = grapePerSqrMKg * areaSqrM * 0.4 / 2.5;

    if (neededWineLtr > producedWineLtr) {
        console.log(`It will be a tough winter! More ${Math.floor(neededWineLtr - producedWineLtr)} liters wine needed.`)
    } else {
        console.log(`Good harvest this year! Total wine: ${Math.floor(producedWineLtr)} liters.`)
        console.log(`${Math.ceil(producedWineLtr - neededWineLtr)} liters left -> ${Math.ceil((producedWineLtr - neededWineLtr) / numberWorkers)} liters per person.`)
    }
}

harvest(["1020",
    "1.5",
    "425",
    "4"
]

)

File: Java_Basic_Jan_2023_SolutionsOfProblems/P02AdditionalProblems_ConditionalStatements/P03_Harvest.java
package P02AdditionalProblems_ConditionalStatements;

import java.util.Scanner;

public class P03_Harvest {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int xScrM = Integer.parseInt(scanner.nextLine());
        double yKgM = Double.parseDouble(scanner.nextLine());
        int zLtr = Integer.parseInt(scanner.nextLine());
        int workers = Integer.parseInt(scanner.nextLine());

        double vineLtr = (xScrM * yKgM) * 0.4 / 2.5;

        if (vineLtr < zLtr) {
            System.out.printf("It will be a tough winter! More %.0f liters wine needed.",
                    Math.floor(zLtr - vineLtr));
        } else {
            System.out.printf("Good harvest this year! Total wine: %.0f liters.%n",
                    Math.floor(vineLtr));
            System.out.printf("%.0f liters left -> %.0f liters per person.",
                    Math.ceil(vineLtr - zLtr), Math.ceil((vineLtr - zLtr) / workers));
        }
    }
}
